### DISCUSSION TITLE
Using readonly (or val) for local variables

### DISCUSSION THREAD

**Discussion by Hixon10** [dotnet/csharplang#7074] [Language Ideas] [answered]
In many cases you don't want to change a local variable, once you assign a value to it. Many languages allow to explicitly say about this intention in the code. 

**Java:**
```java
public static void main(String[] args) {
    final var readonlyVariable = 42;
    final int anotherReadonlyVariable = 43;
}
```

**Kotlin**:
```kotlin
val goldenRatio = 1.618
```

**Rust** goes beyond that. It gives immutability by default, and you need explicitly mark a variable with `mut`:
```rust
fn main() {
    let x = 5;
}
```

So, I would like to have `readonly`, or `val` in C# as well.

**Comment by HaloFour** [comment, marked as answer]
See: #188

**Reply by Hixon10** [reply]
I am sorry.
